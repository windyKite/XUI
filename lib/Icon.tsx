import React from "react";
import wechat from "./icons/wechat.svg";
interface IconProps {
  name: string;
}

const Icon: React.FunctionComponent<IconProps> = props => {
  return <span>{props.name}</span>;
};

export default Icon;
