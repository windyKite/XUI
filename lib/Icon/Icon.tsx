import React from "react";
import "./importIcons";
import "./Icon.scss";
import classNames from "../helpers/classnames";

interface IconProps extends React.SVGAttributes<SVGElement> {
  name: string;
}

const Icon: React.FunctionComponent<IconProps> = ({
  className,
  name,
  ...restProps
}) => {
  return (
    <svg className={classNames("Xui-icon", className)} {...restProps}>
      <use xlinkHref={`#${name}`} />
    </svg>
  );
};

export default Icon;
