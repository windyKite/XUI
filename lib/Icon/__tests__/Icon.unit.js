import * as renderer from 'react-test-renderer'
import React from 'react'
import Icon from '../Icon'

describe('Icon', () => {
    it('是个 Icon', () => {
        const json = renderer.create(<Icon name="wechat"/>).toJSON()
        expect(json).toMatchSnapshot()
    })
}) 