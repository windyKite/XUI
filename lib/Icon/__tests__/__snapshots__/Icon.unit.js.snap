// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon 是个 Icon 1`] = `
<svg
  className="Xui-icon"
>
  <use
    xlinkHref="#wechat"
  />
</svg>
`;
